jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -122,8 +123,11 @@ public void run() {
 			}
 		}, 1000, 1000);
 
-		grid.getPlayer(player).setMessenger(connector);
-		clock.setTaskExecutor(new GuiTaskExecutor(grid, this, playerId));
+		Player player = grid.getPlayer(playerId);
+		if(player != null ){
```

Comment:
This can happen in the tests. There, the GuiInterface is not always initialized with an existing player (which is no problem for the tests). Before my changes, all players where initialized in the PartitionsGrid. However, as this has some downsides, I changed it there to only initialize the existing players. That's why this check became necessary.